store.go

```diff
@@ -2479,13 +2479,29 @@ func (s *store) Mount(id, mountLabel string) (string, error) {
 	if err != nil {
 		return "", err
 	}
+
+	s.graphLock.Lock()
+	defer s.graphLock.Unlock()
+
 	rlstore.Lock()
 	defer rlstore.Unlock()
 	if modified, err := rlstore.Modified(); modified || err != nil {
 		if err = rlstore.Load(); err != nil {
 			return "", err
 		}
 	}
+
+	/* We need to make sure the home mount is present when the Mount is done.  */
+	if s.graphLock.TouchedSince(s.lastLoaded) {
```

Comment:
We probably want to refactor this into an own method because it’s used in multiple places in this file. :)

Reply:
I had refactored the code in a single function in one of the versions I've pushed yesterday but I reverted the change as the new block is different than other similar cases as it also re-initialize the graphDriver